src/rgw/rgw_data_sync.cc

```diff
@@ -5341,20 +5366,15 @@ int RGWSyncBucketCR::operate(const DoutPrefixProvider *dpp)
         yield call(new ReadCR(dpp, env->async_rados, env->svc->sysobj,
                             status_obj, &bucket_status, false, &objv));
         if (retcode < 0) {
-          RELEASE_LOCK(bucket_lease_cr);	
+          RELEASE_LOCK(bucket_lease_cr);
           tn->log(20, SSTR("ERROR: reading the status after acquiring the lock failed. error: " << retcode));
           return set_cr_error(retcode);
         }
         tn->log(20, SSTR("status after acquiring the lock is: " << bucket_status.state));
 
-        // init sync status
-        yield {
-          init_check_compat = objv.read_version.ver <= 1; // newly-created
-          const int num_shards = sync_pipe.dest_bucket_info.layout.current_index.layout.normal.num_shards;
-          call(new InitBucketFullSyncStatusCR(sc, sync_pair, status_obj,
-                                              bucket_status, objv, num_shards,
-                                              init_check_compat, info));
-        }
```

Comment:
`sync_pipe` here has access to both the source- and dest- bucket info. you could pass in a reference to `sync_pipe.source_bucket_info` to avoid re-reading it in `InitBucketFullSyncStatusCR`

Reply:
Done!